pandas/core/internals/blocks.py

```diff
@@ -965,7 +965,13 @@ def setitem(self, indexer, value):
             # GH25495 - If the current dtype is not categorical,
             # we need to create a new categorical block
             values[indexer] = value
-            return self.make_block(Categorical(self.values, dtype=arr_value.dtype))
+            if values.ndim == 2:
```

Comment:
you could instead do
```values = np.atleast_2d(values)[:, 0]``` generically i think

Reply:
we dont get a whole lot of mileage out of that bc we still need to (well sort of, it shouldnt be reached) exclude the raising case on L972